external/fv3fit/fv3fit/pytorch/cyclegan/cyclegan_trainer.py

```diff
@@ -224,6 +225,163 @@ def get_r2(predicted, target) -> float:
     return 1.0 - np.var(predicted - target) / np.var(target)
 
 
+def _hist_multi_channel(array: np.ndarray, bins: np.ndarray):
+    """
+    Compute a histogram for each channel of a multi-channel array.
+
+    Args:
+        array: array of shape (n_samples, n_tiles, n_channels, ...)
+        bins: array of shape (n_bins + 1,)
+
+    Returns:
+        array of shape (n_channels, n_bins)
+    """
+    out_hist = np.empty((array.shape[2], bins.shape[0] - 1), dtype=np.float64)
+    for i_channel in range(array.shape[2]):
+        out_hist[i_channel, :] = np.histogram(
+            array[:, :, i_channel].flatten(), bins=bins, density=True
+        )[0]
+    return out_hist
+
+
+class ResultsAggregator:
+    def __init__(self, histogram_vmax: float):
+        self._total_fake_a: Optional[np.ndarray] = None
```

Comment:
This would definitely make it cleaner and I was about to update it this way, but I realized there's a performance penalty (new array allocation) to doing `x = x + update` over `x += update` which updates `x` in-place without allocating new memory. I don't know whether this performance matters (it's not really worth the time to test, but it is called for every sample of every batch), but given the code's already written and copilot makes it fairly easy to write, I think I'll side on leaving it.